cocos/3d/CCMesh.cpp

```diff
@@ -613,8 +613,8 @@ void Mesh::setLightUniforms(Pass* pass, Scene* scene, const Vec4& color, unsigne
         if (hasAmbient)
         {
             ambient.x /= 255.f; ambient.y /= 255.f; ambient.z /= 255.f;
+            glProgramState->setUniformVec4("u_color", Vec4(color.x * ambient.x, color.y * ambient.y, color.z * ambient.z, color.w));
```

Comment:
setLightUniforms take effect only then there is light. We should also set the `u_color` when there is no light exist.